diff --git a/compiler/options.nim b/compiler/options.nim
index ea3b71a56..d7e794945 100644
--- a/compiler/options.nim
+++ b/compiler/options.nim
@@ -755,7 +755,8 @@ proc setDefaultLibpath*(conf: ConfigRef) =
   if conf.libpath.isEmpty:
     # choose default libpath:
     var prefix = getPrefixDir(conf)
-    conf.libpath = prefix / RelativeDir"lib"
+    conf.libpath = prefix / RelativeDir"lib/nim"
+    if not dirExists(conf.libpath.string): conf.libpath = prefix / RelativeDir"lib"
 
     # Special rule to support other tools (nimble) which import the compiler
     # modules and make use of them.
